spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -99,9 +117,6 @@ protected void onInit() {
 	protected void doStart() {
 		Assert.state(this.outputChannel != null || StringUtils.hasText(this.outputChannelName),
 				"'outputChannel' or 'outputChannelName' is required");
```

Comment:
I completely understand, I just don't want to put the assert in the main path (on every send).